src/GroupManager.php

```diff
@@ -230,11 +268,85 @@ public function removeGroup($entity_type_id, $bundle_id) {
 
       // Only update and refresh the map if a key was found and unset.
       $editable->set('groups', $groups);
-      $saved = $editable->save();
+      $editable->save();
+
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
 
       $this->resetGroupMap();
+    }
+  }
+
+  /**
+   * Creates default roles for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to create default roles.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to create default roles.
+   *
+   * @todo: Would a dedicated RoleManager service be a better place for this?
+   */
+  protected function createRoles($entity_type_id, $bundle_id) {
```

Comment:
maybe `createPerBundleRoles()` - as we'll soon need to add an option also to `createPerGroupRoles()` - for group that want to manage their own roles and permissions settings?